### PR TITLE
MEN-2359: Update mender-convert docs

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

File: 03.Devices/03.Debian-family/docs.md [hunk 1]
```diff
@@ -4,7 +4,39 @@ taxonomy:
...
-This section describes the requirements for Mender when integrated
-with the Debian family target OS images such as
-[Debian](https://www.debian.org/?target=_blank), [Ubuntu](https://www.ubuntu.com/?target=_blank)
-and [Raspbian](https://www.raspberrypi.org/downloads/raspbian/?target=_blank).
+This page describes the requirements for Mender when integrated with the Debian
+family target OS images such as [Debian](https://www.debian.org/?target=_blank),
+[Ubuntu](https://www.ubuntu.com/?target=_blank) and
+[Raspbian](https://www.raspberrypi.org/downloads/raspbian/?target=_blank).
+
+For these devices, `mender-convert` tool is used to perform Mender integration.
```
Comment (estenberg) [inline 03.Devices/03.Debian-family/docs.md]: ... *the* mender-convert tool...

File: 03.Devices/03.Debian-family/docs.md [hunk 2]
```diff
@@ -4,7 +4,39 @@ taxonomy:
...
+
+The tool will take care of the full integration, which includes:
+* Modification of the original partition layout to fulfill Mender needs
+* Integration with a compatible bootloader (see below for details)
+* Install the Mender client and its configuration into the image.
+
+For Debian family, the procedure for Mender integration is the same as for
```
Comment (estenberg) [inline 03.Devices/03.Debian-family/docs.md]: For *the* Debian family of OSes, ...

Comment (estenberg) [reply 03.Devices/03.Debian-family/docs.md]: I think I understand what you mean here (mender-convert creates both .sdimg and .mender files), but I feel like this sentence could be a bit confusing to beginners. Perhaps drop it if not needed?

File: 03.Devices/03.Debian-family/docs.md [hunk 3]
```diff
@@ -4,7 +4,39 @@ taxonomy:
...
+[mender-convert](https://github.com/mendersoftware/mender-convert?target=_blank)
+tools use GRUB as second stage bootloader for Debian on Beaglebone. This tool
+build GRUB from the [official
+repository](https://www.gnu.org/software/grub/grub-download.html?target=_blank).
+Mender does not require any patches for GRUB and should be built with EFI
+platform support.
+
+### U-Boot
+
+The
```
Comment (estenberg) [inline 03.Devices/03.Debian-family/docs.md]: The mender-convert *tool* (no s).

**Comment by estenberg** [comment]
Perhaps ../../artifacts/debian-family should also be updated a bit? It is a bit of a duplicate right now.
Would borrow some text from "An improved workflow to generate Artifacts" in https://hub.mender.io/t/raspberry-pi-3-model-b-b-raspbian/140 for it, perhaps with more concrete steps on the commands to run.

**Comment by estenberg** [comment]
Also, would have a link from the end of device integration to how to generate artifacts, which is the next step.

File: 03.Devices/03.Debian-family/docs.md (hunk 2 above)
Comment (lluiscampos) [reply 03.Devices/03.Debian-family/docs.md]: Mmm. I can see that the sentence is confusing

The point is that we are redirecting the user to a page that talks about generating artifacts, while he/she is looking for how to integrate mender in a Debian SD image. Hence, I think we need to clarify here that following the Artifact generation instructions will also lead to a full image.

**Comment by lluiscampos** [comment]
> Also, would have a link from the end of device integration to how to generate artifacts, which is the next step.

I see your point now. This link is what I tried to have above (refer to our other discussion). So I will rework it to have it at the end, instead, and see if it makes more sense to you.

**Comment by lluiscampos** [comment]
Second round of changes ready.

File: 03.Devices/03.Debian-family/docs.md [hunk 4]
```diff
@@ -4,7 +4,42 @@ taxonomy:
...
+* Modification of the original partition layout to fulfill Mender needs
+* Integration with a compatible bootloader (see below for details)
+* Install the Mender client and its configuration into the image.
+
+##Bootloader support
+
+### GRUB
+
+The
+[mender-convert](https://github.com/mendersoftware/mender-convert?target=_blank)
```
Comment (estenberg) [inline 03.Devices/03.Debian-family/docs.md]: ...build*s* GRUB...

File: 03.Devices/03.Debian-family/docs.md [hunk 5]
```diff
@@ -4,7 +4,42 @@ taxonomy:
...
+tool provides for building and installing [patched
+U-Boot](https://github.com/mendersoftware/uboot-mender?target=_blank) for
+Raspbian. Currently only a patched U-Boot is supported by Mender on Raspberry Pi
+3.
+
+## Mender integration
+
+The procedure to be followed to integrate Mender is the same as for to create
```
Comment (estenberg) [inline 03.Devices/03.Debian-family/docs.md]: ...create a Mender Artifact..

File: 04.Artifacts/02.Debian-family/docs.md [hunk 6]
```diff
@@ -6,18 +6,37 @@ taxonomy:
...
+
+The recommended workflow for using `mender-convert` is to rely on a *golden disk
+image* that should be replicated in a robust way to many devices. The steps in
+this workflow are:
+
+1. Install a fresh OS to a device storage (typically an SD card)
+2. Boot the device, log in, and make modifications in run-time, e.g. install
+   packages, change configurations
+3. Power off the device with the (now updated) golden image
+5. Copy the contents of the golden image back to your workstation (e.g. using
```
Comment (estenberg) [inline 04.Artifacts/02.Debian-family/docs.md]: 4?

File: 04.Artifacts/02.Debian-family/docs.md [hunk 7]
```diff
@@ -6,18 +6,37 @@ taxonomy:
...
+modified during deployments. It is simply the "source" for generating the
+Artifacts that you deploy to the devices in the field.
+
+In order to create another OTA update, the device with the golden image is
+booted again, i.e. the steps from 2. and onwards are carried out again.
 
-!!! `mender-convert` is currently tested on BeagleBone and Raspberry Pi3, using official Debian or Raspbian images. The intention is to extend and test the utility to cover more boards and OSes and finally make it board-agnostic.
+!!! `mender-convert` is currently tested on BeagleBone and Raspberry Pi3, using
+official Debian or Raspbian images. The intention is to extend and test the
+utility to cover more boards and OSes and finally make it board-agnostic.
```
Comment (estenberg) [inline 04.Artifacts/02.Debian-family/docs.md]: utility or tool?

File: 03.Devices/03.Debian-family/docs.md (hunk 5 above)
Comment (estenberg) [reply 03.Devices/03.Debian-family/docs.md]: Should this note with links how to create Artifacts be moved to the end?

File: 03.Devices/03.Debian-family/docs.md [hunk 8]
```diff
@@ -4,7 +4,42 @@ taxonomy:
...
+Mender does not require any patches for GRUB and should be built with EFI
+platform support.
+
+### U-Boot
+
+The
+[mender-convert](https://github.com/mendersoftware/mender-convert?target=_blank)
+tool provides for building and installing [patched
+U-Boot](https://github.com/mendersoftware/uboot-mender?target=_blank) for
+Raspbian. Currently only a patched U-Boot is supported by Mender on Raspberry Pi
```
Comment (apodogrocki) [inline 03.Devices/03.Debian-family/docs.md]: What about Raspberry Pi 0 W? Although there is an issue with partition resizing now, but anyway patched U-Boot is also used there...

File: 04.Artifacts/02.Debian-family/docs.md [hunk 9]
```diff
@@ -6,18 +6,37 @@ taxonomy:
...
 
-A typical workflow for using `mender-convert` is to rely on a *golden disk image* that should be replicated in a robust way to many devices. The steps in this workflow are:
-
-1. Install a fresh OS to a device storage
-2. Boot the device
-3. Make modifications in run-time, e.g. install packages, change configurations
-4. Power off the device with the (now updated) golden image
-5. Generate a Mender Artifact and disk image from this golden image using `mender-convert`
+The output of the tool is:
+* The converted disk image described above ready to be flashed to as new device.
```
Comment (apodogrocki) [inline 04.Artifacts/02.Debian-family/docs.md]: Maybe: The converted disk image described above ready to be flashed onto SD card.

File: 04.Artifacts/02.Debian-family/docs.md [hunk 10]
```diff
@@ -6,18 +6,37 @@ taxonomy:
...
+* A Mender Artifact formatted updated to be deployed remotely to devices running
+  Mender through the Mender Server.
+
```
Comment (apodogrocki) [inline 04.Artifacts/02.Debian-family/docs.md]: @lluiscampos ...and ext4 image

File: 03.Devices/03.Debian-family/docs.md (hunk 5 above)
Comment (lluiscampos) [reply 03.Devices/03.Debian-family/docs.md]: It is at the end.

File: 04.Artifacts/02.Debian-family/docs.md (hunk 10 above)
Comment (lluiscampos) [reply 04.Artifacts/02.Debian-family/docs.md]: What is the purpose of the ext4 image? What is it used for?

Comment (lluiscampos) [reply 04.Artifacts/02.Debian-family/docs.md]: I am adding the following. Please suggest a rewrite if required:

* An ext4 image of the modified filesystem for advance users.

File: 04.Artifacts/02.Debian-family/docs.md (hunk 9 above)
Comment (lluiscampos) [reply 04.Artifacts/02.Debian-family/docs.md]: :+1:

File: 03.Devices/03.Debian-family/docs.md (hunk 8 above)
Comment (lluiscampos) [reply 03.Devices/03.Debian-family/docs.md]: Here and in the README of mender-convert, we are not clearly stating that Raspberry Pi Zero W is supported. I think it is better to wait until we resolve the issue about the resize (currently WIP by you) and then we update the documentation in both places

**Comment by lluiscampos** [comment]
Third round ready.

@estenberg Thanks for the tip on tool/utility. I replaced everywhere to use only `mender-convert`.

File: 04.Artifacts/02.Debian-family/docs.md (hunk 10 above)
Comment (estenberg) [reply 04.Artifacts/02.Debian-family/docs.md]: advance -> advanced

Comment (lluiscampos) [reply 04.Artifacts/02.Debian-family/docs.md]: :+1:

**Comment by lluiscampos** [comment]
Fixed the last typo.

Thanks for the review. Merging!

**Pull request merged**